src/ligo.ml

```diff
@@ -240,46 +240,94 @@ let geq_timestamp_timestamp = Z.geq
 
 (* tez *)
 
-type tez = Z.t
+type tez = Int64.t
 
 let tez_from_literal s =
   try
     let n = parse_int_with_suffix "mutez" s in
     if Z.lt n Z.zero then
-      failwith "Ligo.tez_from_literal: negative"
-    else n
+      failwith "Ligo.tez_from_literal: underflow"
```

Comment:
I'm not sure if "underflow" or simply "negative" as it was before would make more sense in this message :thinking:

Reply:
Ah, true. Since this is creation perhaps "negative" is more appropriate. Good point! :+1:

Reply:
Changed them both since I like me some symmetry :smile::
```ocaml
    [..]
    if Z.lt n Z.zero then
      failwith "Ligo.tez_from_literal: out of range (negative)"
    else if Z.gt n (Z.of_int64 Int64.max_int) then
      failwith "Ligo.tez_from_literal: out of range (positive)"
    else
    [..]
```